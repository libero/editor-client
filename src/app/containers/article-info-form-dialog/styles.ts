import { makeStyles } from '@material-ui/core/styles';

export const useArticleInfoFormStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(1, 2.5, 0, 2.5)
  },

  inputField: {
    marginBottom: theme.spacing(2.5)
  },

  copyrightStatement: {
    paddingBottom: theme.spacing(2),
    paddingTop: theme.spacing(2)
  },

  buttonPanel: {
    display: 'flex',
    justifyContent: 'flex-end',
    marginTop: theme.spacing(4),
    '& > *:last-child': {
      marginLeft: theme.spacing(2)
    }
  },
  spacer: {
    flexGrow: 1
  }
}));
