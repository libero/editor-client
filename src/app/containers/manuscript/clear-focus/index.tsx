import React, { useCallback } from 'react';
import { useDispatch } from 'react-redux';
import * as manuscriptEditorAction from 'app/actions/manuscript-editor.actions';

export const ClearFocus: React.FC<{ className?: string }> = ({ children, className }) => {
  const dispatch = useDispatch();
  const handleClick = useCallback(() => {
    dispatch(manuscriptEditorAction.removeFocusAction());
  }, [dispatch]);

  return (
    <div className={className} onClick={handleClick}>
      {children}
    </div>
  );
};
