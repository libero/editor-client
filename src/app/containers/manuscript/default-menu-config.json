{
  "button-group": [
    "save",
    "undo",
    "redo",
    "pdf"
  ],
  "menus": [
    "headings",
    "typography",
    "insert"
  ],
  "custom-buttons": []
}