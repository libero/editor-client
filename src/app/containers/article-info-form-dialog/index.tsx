import React, { useCallback, SyntheticEvent, useState, ChangeEvent } from 'react';
import { TextField } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import { isEqual, set } from 'lodash';
import Interweave from 'interweave';

import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import * as manuscriptActions from 'app/actions/manuscript.actions';
import { ActionButton } from 'app/components/action-button';
import { useArticleInfoFormStyles } from 'app/containers/article-info-form-dialog/styles';
import { getArticleInformation, getAuthors } from 'app/selectors/manuscript.selectors';
import formGrid from 'app/styles/form-grid.module.scss';
import { Select } from 'app/components/select';
import {
  ArticleInformation,
  getCopyrightStatement,
  getLicenseTextEditorState,
  LICENSE_CC0,
  LICENSE_CC_BY_4
} from 'app/models/article-information';
import { SectionContainer } from 'app/components/section-container';
import { stringifyEditorState } from 'app/utils/view.utils';

const labelProps = { shrink: true };

const SubjectOptions = [
  'Biochemistry and Chemical Biology',
  'Cancer Biology',
  'Cell Biology',
  'Chromosomes and Gene Expression',
  'Computational and Systems Biology',
  'Developmental Biology',
  'Ecology',
  'Epidemiology and Global Health',
  'Evolutionary Biology',
  'Genetics and Genomics',
  'Medicine',
  'Immunology and Inflammation',
  'Microbiology and Infectious Disease',
  'Neuroscience',
  'Physics of Living Systems',
  'Plant Biology',
  'Stem Cells and Regenerative Medicine',
  'Structural Biology and Molecular Biophysics'
].map((title) => ({ label: title, value: title }));

export const ArticleInfoFormDialog: React.FC<{}> = () => {
  const classes = useArticleInfoFormStyles();
  const dispatch = useDispatch();
  const articleInfo = useSelector(getArticleInformation);

  const [userArticleInfo, setArticleInfo] = useState<ArticleInformation>(articleInfo);
  const authors = useSelector(getAuthors);
  const closeDialog = useCallback(() => {
    dispatch(manuscriptEditorActions.hideModalDialog());
  }, [dispatch]);

  const handleFormChange = useCallback(
    (event: SyntheticEvent) => {
      const fieldName = event.target['name'];
      const newValue = event.target['value'];
      setArticleInfo({
        ...userArticleInfo,
        [fieldName]: newValue
      });
    },
    [userArticleInfo, setArticleInfo]
  );

  const handleSubjectChange = useCallback(
    (event: ChangeEvent<{ name: string; value: string }>) => {
      const fieldName = event.target['name'];
      const newValue = event.target['value'];
      const updatedArticleInfo = {
        ...userArticleInfo,
        subjects: [...userArticleInfo.subjects]
      };

      set(updatedArticleInfo, fieldName, newValue);
      updatedArticleInfo.subjects = updatedArticleInfo.subjects.filter(Boolean);
      setArticleInfo(updatedArticleInfo);
    },
    [userArticleInfo, setArticleInfo]
  );

  const handleLicenseTypeChange = useCallback(
    (event: ChangeEvent<{ name: string; value: string }>) => {
      const newLicenseType = event.target['value'];
      const updatedArticleInfo = {
        ...userArticleInfo,
        licenseType: newLicenseType,
        licenseText: getLicenseTextEditorState(newLicenseType),
        copyrightStatement:
          newLicenseType === LICENSE_CC_BY_4 ? getCopyrightStatement(authors, userArticleInfo.publicationDate) : ''
      };
      setArticleInfo(updatedArticleInfo);
    },
    [userArticleInfo, authors]
  );

  const handleAccept = useCallback(() => {
    if (!isEqual(userArticleInfo, articleInfo)) {
      dispatch(manuscriptActions.updateArticleInformationAction(userArticleInfo));
    }
    closeDialog();
  }, [closeDialog, articleInfo, userArticleInfo, dispatch]);

  return (
    <section className={classes.root}>
      <div className={formGrid.container}>
        <Select
          className={formGrid.fullWidth}
          name="subjects.0"
          placeholder="Please select"
          fullWidth
          canUnselect={true}
          blankValue={undefined}
          label="Subject *"
          value={userArticleInfo.subjects[0]}
          onChange={handleSubjectChange}
          options={SubjectOptions}
        />
        <Select
          className={formGrid.fullWidth}
          name="subjects.1"
          placeholder="Please select"
          fullWidth
          canUnselect={true}
          blankValue={undefined}
          label="Subject"
          value={userArticleInfo.subjects[1]}
          onChange={handleSubjectChange}
          options={SubjectOptions}
        />
        <TextField
          fullWidth
          name="articleDOI"
          label="Article DOI"
          classes={{ root: formGrid.fullWidth }}
          InputLabelProps={labelProps}
          variant="outlined"
          value={userArticleInfo.articleDOI}
          onChange={handleFormChange}
        />
        <TextField
          fullWidth
          name="publisherId"
          label="Publisher ID"
          classes={{ root: formGrid.fullWidth }}
          InputLabelProps={labelProps}
          variant="outlined"
          value={userArticleInfo.publisherId}
          onChange={handleFormChange}
        />
        <TextField
          fullWidth
          name="elocationId"
          label="eLocation ID"
          classes={{ root: formGrid.fullWidth }}
          InputLabelProps={labelProps}
          variant="outlined"
          value={userArticleInfo.elocationId}
          onChange={handleFormChange}
        />
        <TextField
          fullWidth
          name="volume"
          label="Volume"
          classes={{ root: formGrid.firstCol }}
          InputLabelProps={labelProps}
          variant="outlined"
          value={userArticleInfo.volume}
          onChange={handleFormChange}
        />
        <TextField
          fullWidth
          name="publicationDate"
          label="Published"
          type="date"
          classes={{ root: formGrid.fullWidth }}
          InputLabelProps={labelProps}
          variant="outlined"
          value={userArticleInfo.publicationDate}
          onChange={handleFormChange}
        />
        <Select
          className={formGrid.fullWidth}
          placeholder="Please select"
          fullWidth
          blankValue={undefined}
          label="License type"
          value={userArticleInfo.licenseType}
          onChange={handleLicenseTypeChange}
          options={[
            { label: LICENSE_CC_BY_4, value: LICENSE_CC_BY_4 },
            { label: LICENSE_CC0, value: LICENSE_CC0 }
          ]}
        />
        {userArticleInfo.licenseType === LICENSE_CC_BY_4 ? (
          <SectionContainer label="Copyright statement" variant="outlined" className={formGrid.fullWidth}>
            {userArticleInfo.copyrightStatement}
          </SectionContainer>
        ) : undefined}
        <SectionContainer label="Permissions" variant="outlined" className={formGrid.fullWidth}>
          <Interweave content={stringifyEditorState(userArticleInfo.licenseText)} />
        </SectionContainer>
      </div>
      <div className={classes.buttonPanel}>
        <ActionButton variant="secondaryOutlined" onClick={closeDialog} title="Cancel" />
        <ActionButton variant="primaryContained" onClick={handleAccept} title="Done" />
      </div>
    </section>
  );
};
