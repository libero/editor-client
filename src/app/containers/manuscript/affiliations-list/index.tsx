import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { IconButton } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';

import { SectionContainer } from 'app/components/section-container';
import { ActionButton } from 'app/components/action-button';
import { getAffiliations } from 'app/selectors/manuscript.selectors';
import { Affiliation } from 'app/models/affiliation';
import { useAffiliationStyles } from './styles';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import { ConnectedAffiliationFormDialog } from 'app/containers/affiliation-form-dialog';
import { ComponentWithId } from 'app/types/utility.types';

export const AffiliationsList: React.FC<ComponentWithId> = ({ id }) => {
  const classes = useAffiliationStyles();
  const affiliations = useSelector(getAffiliations);
  const dispatch = useDispatch();

  const editAffiliation = useCallback(
    (aff: Affiliation) => {
      dispatch(
        manuscriptEditorActions.showModalDialog({
          component: ConnectedAffiliationFormDialog,
          props: { affiliation: aff },
          title: 'Edit Affiliation'
        })
      );
    },
    [dispatch]
  );

  const addAffiliation = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: ConnectedAffiliationFormDialog,
        props: {},
        title: 'Add Affiliation'
      })
    );
  }, [dispatch]);

  const renderAffiliation = (aff: Affiliation): React.ReactNode => (
    <li key={aff.id} className={classes.listItem}>
      <div className={classes.orderLabel}>{affiliations.length > 1 ? `(${aff.label})` : undefined}</div>
      <div className={classes.affiliationInfo}>{aff.getDisplayName()}</div>
      <IconButton classes={{ root: classes.editButton }} onClick={editAffiliation.bind(null, aff)}>
        <EditIcon fontSize="small" />
      </IconButton>
    </li>
  );

  return (
    <section>
      <SectionContainer label="Affiliations" id={id}>
        <ul className={classes.root}>{affiliations.map(renderAffiliation)}</ul>
      </SectionContainer>
      <ActionButton
        title="Add Affiliation"
        variant="addEntity"
        onClick={addAffiliation}
        className={classes.addButton}
      />
    </section>
  );
};
