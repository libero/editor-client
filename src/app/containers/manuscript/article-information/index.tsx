import React, { useCallback } from 'react';
import { IconButton } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';
import { useSelector, useDispatch } from 'react-redux';

import { SectionContainer } from 'app/components/section-container';
import { getArticleInformation } from 'app/selectors/manuscript.selectors';
import { useArticleInformationStyles } from 'app/containers/manuscript/article-information/styles';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import { ArticleInfoFormDialog } from 'app/containers/article-info-form-dialog';

export const ArticleInformation: React.FC<{}> = () => {
  const articleInfo = useSelector(getArticleInformation);
  const dispatch = useDispatch();
  const editArticleInfo = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: ArticleInfoFormDialog,
        title: 'Article Information'
      })
    );
  }, [dispatch]);

  const classes = useArticleInformationStyles();
  return (
    <SectionContainer label="Article information" className={classes.sectionContainer}>
      <section className={classes.infoSection}>
        <div>
          <strong> Article DOI: </strong> {articleInfo.articleDOI}
        </div>
        <div>
          <strong> Publisher ID: </strong> {articleInfo.publisherId}
        </div>
      </section>
      <IconButton classes={{ root: classes.editButton }} onClick={editArticleInfo}>
        <EditIcon fontSize="small" />
      </IconButton>
    </SectionContainer>
  );
};
