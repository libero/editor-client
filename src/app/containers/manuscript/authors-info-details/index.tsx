import React, { useCallback } from 'react';
import { IconButton } from '@material-ui/core';
import { useSelector, useDispatch } from 'react-redux';
import EditIcon from '@material-ui/icons/Edit';
import { isBoolean, isEqual } from 'lodash';
import Interweave from 'interweave';

import { SectionContainer } from 'app/components/section-container';
import { getAuthorAffiliations, getAuthors } from 'app/selectors/manuscript.selectors';
import { Person } from 'app/models/person';
import {
  useAuthorDetailStyles,
  useAuthorsDetailsListStyles
} from 'app/containers/manuscript/authors-info-details/styles';
import { ActionButton } from 'app/components/action-button';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import { AuthorFormDialog } from 'app/containers/author-form-dialog';
import { stringifyEditorState } from 'app/utils/view.utils';
import { OrcidIcon } from 'app/assets/icons';
import { ComponentWithId } from 'app/types/utility.types';

export const AuthorsInfoDetails: React.FC<ComponentWithId> = ({ id }) => {
  const authors = useSelector(getAuthors);
  const classes = useAuthorsDetailsListStyles();
  const dispatch = useDispatch();

  const addAuthor = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: AuthorFormDialog,
        title: 'Add Author'
      })
    );
  }, [dispatch]);

  return (
    <section>
      <SectionContainer label="Author Information" id={id}>
        {authors.map((author) => (
          <AuthorInfo key={author.id} author={author} />
        ))}
      </SectionContainer>
      <ActionButton variant="addEntity" title="Add Author" onClick={addAuthor} className={classes.addAuthorButton} />
    </section>
  );
};

interface AuthorInfoProps {
  author: Person;
}

const AuthorInfo: React.FC<AuthorInfoProps> = React.memo(({ author }) => {
  const classes = useAuthorDetailStyles();
  const dispatch = useDispatch();
  const affiliations = useSelector(getAuthorAffiliations)(author);

  const editAuthor = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: AuthorFormDialog,
        props: { author },
        title: 'Edit Author'
      })
    );
  }, [author, dispatch]);

  return (
    <section className={classes.root}>
      <div className={classes.authorInfoContainer}>
        <div className={classes.authorInfoLine}>
          <strong> {author.getDisplayName()} </strong>
        </div>
        <div className={classes.authorInfoLine}>
          {author.isCorrespondingAuthor ? 'Corresponding Author: ' : undefined}
          {author.email}
        </div>
        <div className={classes.authorInfoLine}>
          {affiliations.map((aff) => (
            <div key={aff.id}>{aff.getDisplayName()}</div>
          ))}
        </div>
        <div className={classes.authorInfoLine}>
          {author.isAuthenticated ? <OrcidIcon className={classes.orcidIcon} /> : undefined}{' '}
          <a href={`https://orcid.org/${author.orcid}`} target="_blank" rel="noopener noreferrer">
            {author.orcid}
          </a>
        </div>
        {isBoolean(author.hasCompetingInterest) ? (
          <div className={classes.authorInfoLine}>
            {author.hasCompetingInterest
              ? `Competing interest: ${author.competingInterestStatement}`
              : 'No Competing interest declared'}
          </div>
        ) : undefined}
        {author.bio ? (
          <div className={classes.authorBio}>
            <Interweave content={stringifyEditorState(author.bio)} />
          </div>
        ) : undefined}
      </div>
      <IconButton classes={{ root: classes.editButton }} onClick={editAuthor}>
        <EditIcon fontSize="small" />
      </IconButton>
    </section>
  );
}, isEqual);
