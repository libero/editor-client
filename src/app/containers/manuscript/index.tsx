import React from 'react';
import { useSelector } from 'react-redux';
import { Backdrop, CircularProgress } from '@material-ui/core';

import { isManuscriptLoaded } from '../../selectors/manuscript.selectors';
import './styles.scss';
import { ManuscriptToolbar } from './manuscript-toolbar';
import { ManuscriptEditor } from './manuscript-editor';

const renderBackdrop = (): JSX.Element => (
  <Backdrop open={true}>
    <CircularProgress color="inherit" />
  </Backdrop>
);

export const ManuscriptContainer: React.FC = () => {
  const isLoaded = useSelector(isManuscriptLoaded);
  const renderContent = (): JSX.Element => (
    <React.Fragment>
      <ManuscriptToolbar />
      <ManuscriptEditor />
    </React.Fragment>
  );

  return isLoaded ? renderContent() : renderBackdrop();
};
