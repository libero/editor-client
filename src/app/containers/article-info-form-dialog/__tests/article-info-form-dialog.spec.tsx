import React from 'react';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

import { givenState } from 'app/test-utils/reducer-test-helpers';
import { ArticleInfoFormDialog } from 'app/containers/article-info-form-dialog/index';
import * as manuscriptActions from 'app/actions/manuscript.actions';

jest.mock('@material-ui/core', () => ({
  TextField: () => <div data-cmp="TextField"></div>,
  Button: () => <div data-cmp="Button"></div>
}));

describe('Article Info Form Dialog', () => {
  const mockStore = configureMockStore([]);
  let mockState;

  beforeEach(() => {
    mockState = givenState({
      articleInfo: {
        articleDOI: '',
        dtd: '1',
        articleType: ''
      }
    });
  });

  it('renders edit article information dialog', () => {
    const store = mockStore({
      manuscript: mockState
    });

    const wrapper = create(
      <Provider store={store}>
        <ArticleInfoFormDialog />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches update article info action', () => {
    const store = mockStore({ manuscript: mockState });
    jest.spyOn(store, 'dispatch');

    const wrapper = mount(
      <Provider store={store}>
        <ArticleInfoFormDialog />
      </Provider>
    );

    wrapper.find({ title: 'Done' }).prop('onClick')(mockState.data.present.articleInfo, []);
    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.updateArticleInformationAction(mockState.data.present.articleInfo)
    );
  });
});
