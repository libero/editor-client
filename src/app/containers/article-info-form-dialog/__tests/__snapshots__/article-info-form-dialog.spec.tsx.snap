// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Info Form Dialog renders edit article information dialog 1`] = `
<section
  className="makeStyles-root-1"
>
  <div
    className="container"
  >
    <input
      data-cmp="Select"
      onChange={[Function]}
    />
    <input
      data-cmp="Select"
      onChange={[Function]}
    />
    <div
      data-cmp="TextField"
    />
    <div
      data-cmp="TextField"
    />
    <div
      data-cmp="TextField"
    />
    <div
      data-cmp="TextField"
    />
    <div
      data-cmp="TextField"
    />
    <input
      data-cmp="Select"
      onChange={[Function]}
      value="CC0"
    />
    <div
      className="fullWidth makeStyles-root-6 makeStyles-outlinedVariant-7"
    >
      <label
        className="makeStyles-label-9"
      >
        Permissions
      </label>
      <span />
    </div>
  </div>
  <div
    className="makeStyles-buttonPanel-4"
  >
    <div
      data-cmp="Button"
    />
    <div
      data-cmp="Button"
    />
  </div>
</section>
`;
